project/yxuan0329/proposal.rst

```diff
@@ -0,0 +1,186 @@
+=================
+R-Tree Spatial Search Engine
+=================
+
+
+..   You are encouraged to use this plain-text `reStructuredText
+..   <https://docutils.sourceforge.io/rst.html>`__ format.  :download:`Download
+..   this file. <project_template.rst>`
+
+.. Please do give your project a name that clearly states the subject and is short
+.. enough for people to remember.
+
+.. Additional documents or files may be used.  They should be added in the
+.. repository.
+
+Basic Information
+=================
+
+.. Create a GitHub repository to host your project and add the URL here.  The
+.. GitHub repository has an 'About' field, in which you are encouraged to write a
+.. simple statement (preferably one sentence) to introduce the project.
+
+R-tree is a tree structure designed for efficient spatial data access. It has 
+been widely used in real-world geographic searching because of its exceptional 
+searching performance. In this project, we aim to serve a R-tree library to 
+provide an efficient search for end-users. 
+
+
+Problem to Solve
+================
+
+.. Describe the problem or the set of problems you want to solve.  Include
+.. necessary background information without making it lengthy.
+
+.. Some points may help you organize the problem description:
+
+.. 1. The field or industry of the problem.
+.. 2. The physics and/or the mathematics behind the problem.
+.. 3. The algorithm or numerical method that should be applied for solving the
+..    problem.
+
+In geographic searching application, the users would expect to swiftly receive 
+a set of searching results upon entering the query. In real-world application, 
+rapid access to geospatial information is crucial for the users, as delays can 
+result in inconveniences and increased costs.  
+
+The fundamental concept behind the R-tree is to encapsulate nearby objects 
+related to the searching target within a minimum bounding rectangle. Our goal 
+is to develop an efficient R-tree library for spatial data retrieval that can 
+address large-scale spatial data searches. 
+
+
+Prospective Users
+=================
+
+.. Describe the users of your software and how they will use it.  It is OK to
+.. combine this section with the previous one (`Problem to solve`_).
+
+This project aims to provide to users who seek to build an efficient 
+multi-dimensional searching service.
```

Comment:
How many dimensions do you want to support?  It couples with the test data.

Reply:
For the initial testing, I will generate the data in 2 or 3 dimensions, with (latitude, longitude, and altitude). After that, I intend to expand the dimensionality of the generated data to 4 or 5 dimensions. This expansion is aimed at testing the capacity to support higher-dimensional data scenarios.

Reply:
Think twice before going into more than 3 dimensions.  Only do it if you are sure it is useful.